src/scribe_data/language_data_extraction/Hindustani/Hindi/emoji_keywords/generate_emoji_keywords.py

```diff
@@ -1,47 +1,57 @@
 """
-Generates keyword-emoji relationships from a selection of Hindi words, ensuring Urdu words are excluded.
-
-.. raw:: html
-    <!--
-    * Copyright (C) 2024 Scribe
-    *
-    * This program is free software: you can redistribute it and/or modify
-    * it under the terms of the GNU General Public License as published by
-    * the Free Software Foundation, either version 3 of the License, or
-    * (at your option) any later version.
-    *
-    * This program is distributed in the hope that it will be useful,
-    * but WITHOUT ANY WARRANTY; without even the implied warranty of
-    * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-    * GNU General Public License for more details.
-    *
-    * You should have received a copy of the GNU General Public License
-    * along with this program.  If not, see <https://www.gnu.org/licenses/>.
-    -->
+* Copyright (C) 2024 Scribe
+*
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <https://www.gnu.org/licenses/>.
 """
 
 import argparse
+from scribe_data.unicode.generate_emoji_keyword import generate_emoji_keyword
 
-from scribe_data.unicode.process_unicode import gen_emoji_lexicon
-from scribe_data.utils import export_formatted_data
-
-LANGUAGE = "Hindi"
-DATA_TYPE = "emoji-keywords"
+# Define the main language
+LANGUAGE = "Hindustani"  # Change to a grouped language if needed
 emojis_per_keyword = 3
 
-parser = argparse.ArgumentParser()
-parser.add_argument("--file-path")
+# Set up the argument parser
+parser = argparse.ArgumentParser(
+    description="Generate emoji keywords for a specific language."
+)
+parser.add_argument(
+    "--file-path", required=True, help="Path to save the generated emoji keywords."
+)
+parser.add_argument(
+    "--sub-languages",
+    nargs="*",
+    help="List of specific sub-languages to process (e.g., Hindi Urdu). If omitted, all sub-languages will be processed.",
+)
+parser.add_argument(
+    "--gender",
+    choices=["male", "female", "neutral"],
+    help="Specify the gender for emoji customization.",
+)
+parser.add_argument(
+    "--region", help="Specify the region for emoji customization (e.g., US, IN)."
+)
+
+# Parse the command-line arguments
```

Comment:
Thanks @catreedle I will take note

Reply:
welcome! I'm not really sure of the differences between individual languages to decide if it is plausible, but it is something to consider. :)

Reply:
@catreedle this will mean to create a modular function  for the common parser file, most likely in a separate file that can be imported to each `generate_emoji_keyword` file for all language. Also @SethiShreya mention the is a parser function somewhere within the codebase, I have not seen it. Please you can help point it out for me study. So I will create another file in the Unicode folder for this parser function related to `generate_emoji_keyword` for each language

Reply:
I can't find it either...
maybe something like this could work?
```py
def setup_argument_parser():
    parser = argparse.ArgumentParser(description="Generate emoji keywords for a specific language.")
    parser.add_argument(
        "--file-path", required=True, help="Path to save the generated emoji keywords."
    )
    ...
    
    
    return parser
```